Dark.mode

kovallux opened the pull request:
About my code. 

I made obj-c category on UIColor, which works very much like class extension in Swift, and added "adaptive" to all color declarations inside framework. 

Also I added a universal converter for all colors, it reduces brightness of color. 
But I also added another method 

```objc
-(UIColor* _Nonnull)adaptiveWithDark:(UIColor* _Nonnull)color; 
```

Use it to explicitly define color variant for dark mode if universal converter is not good for specific color. You can find example if its usage in demo, see CTApperanceViewController, line 62.

kovallux committed:
branch test, gitignore update

kovallux committed:
update podfile to new syntax

kovallux committed:
fixed errors

kovallux committed:
add new category to pod

kovallux committed:
add adaptive category to all color in library

kovallux committed:
add adaptive code to defines

kovallux committed:
add cartfile

kovallux committed:
change color of white label

Patrick-Kladek commented on .gitignore:
```diff
@@ -61,6 +61,7 @@ xcuserdata/
 # https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control
 
 Pods/
+Podfile.lock
```
according to https://guides.cocoapods.org/using/using-cocoapods.html PodFile.lock should not be ignored: "the Podfile and Podfile.lock should always be kept under version control."

Please revert this.

Patrick-Kladek commented on CTAssetsPickerController/CTAssetCollectionViewCell.h:
```diff
@@ -48,7 +48,7 @@
 
 
-- (instancetype)initWithThumbnailSize:(CGSize)size reuseIdentifier:(nullable NSString *)reuseIdentifier;
+- (nullable instancetype)initWithThumbnailSize:(CGSize)size reuseIdentifier:(nullable NSString *)reuseIdentifier;
```
Thanks for adding annotations 👍

Patrick-Kladek commented on CTAssetsPickerController/Categories/DarkMode/UIColor+SKDarkMode.m:
```diff
@@ -0,0 +1,111 @@
+//                        newSaturation = 0.5 + diff;
+//                    }
+//
+//                    return [UIColor colorWithHue:hue saturation:newSaturation brightness:brightness alpha:alpha];
```
Please remove unused code

Patrick-Kladek commented on Podfile:
```diff
@@ -1,7 +1,11 @@
+use_frameworks!
+
+target 'CTAssetsPickerDemo' do
+  pod 'CTAssetsPickerController', :path => '.'
```
I suspect this was necessary during testing right? If so please revert this as I don't want to break existing CocoaPods support.

kovallux replied:
This is necessary for all pods since several years. It's just a new way of writing pod file. If you run pod init on new project, you will see target statement in new podfile. It does not break existing cocoapod.

<img width="944" alt="Screenshot 2020-04-22 at 08 39 15" src="https://user-images.githubusercontent.com/3957568/79945042-ad320d00-8475-11ea-8362-d905a8ed4792.png">

Patrick-Kladek commented on CTAssetsPickerController/CTAssetsPickerDefines.h:
```diff
@@ -35,35 +35,35 @@
 
 #define CTAssetCollectionViewCellTitleFont          [UIFont preferredFontForTextStyle:UIFontTextStyleBody]
-#define CTAssetCollectionViewCellTitleTextColor     [UIColor darkTextColor]
+#define CTAssetCollectionViewCellTitleTextColor     [[UIColor darkTextColor] adaptive]
```
The `adaptive` approach was definitely the easiest & fastest solution :)

Patrick-Kladek reviewed (changes requested):
Looks good overall, please fix these minor issues

kovallux committed:
change cartfile

kovallux committed:
fixes per pull request comments

kovallux commented:
Hi Patrick, I merged latest fixes, please revise PR.

Patrick-Kladek committed:
add CTAssetPickerController framework target

Patrick-Kladek committed:
Move scheme to workspace

Patrick-Kladek committed:
Fix typo in scheme

Patrick-Kladek committed:
Restructure Project

- Remove CocoaPods entirely
- Rename Targets
- Kind of hack together umbrella header
- Fix Bundle Lookup other CTAssetsPickerLocalizedString() returned nil

Patrick-Kladek committed:
update gitignore

Patrick-Kladek committed:
Add PureLayout Build Output

Patrick-Kladek committed:
Remove demo scheme

Patrick-Kladek committed:
bump version to iOS 10

Patrick-Kladek committed:
use private cartfile

Patrick-Kladek committed:
Rename framework to CTAssetsPicker & include umbrella header

Patrick-Kladek committed:
Rename project

Patrick-Kladek committed:
Rename folder

Patrick-Kladek committed:
cleanup xcode files from repo

Patrick-Kladek committed:
Add shared scheme to repo

Patrick-Kladek committed:
Fix warnings

Patrick-Kladek committed:
Merge remote-tracking branch 'refs/remotes/origin/dark.mode'

Conflicts:
	.gitignore

Patrick-Kladek committed:
Remove public cartfile

Patrick-Kladek committed:
Don't embed PureLayout inside CTAssetsPicker framework

So called Umbrella frameworks are not supported on iOS
https://stackoverflow.com/a/49039474

Patrick-Kladek committed:
get rid of NSLog statement

Patrick-Kladek committed:
Purposely make duration label white during dark/light mode

Patrick-Kladek reviewed (approved):
Thanks for fixing the missing points

Patrick-Kladek committed:
Move classes to own folder

Patrick-Kladek committed:
Code style

Patrick-Kladek committed:
make CTAssetsPickerController public

Patrick-Kladek committed:
Merge pull request #2 from medbee/scratchpad/carthage

Carthage Support
